tools/istio-iptables/pkg/cmd/run_test.go

```diff
@@ -0,0 +1,360 @@
+		"ip6tables -t nat -A ISTIO_OUTPUT -o lo ! -d ::1/128 -j ISTIO_IN_REDIRECT",
+		"ip6tables -t nat -A ISTIO_OUTPUT -d ::1/128 -j RETURN",
+		"ip6tables -t nat -A ISTIO_OUTPUT -j ISTIO_REDIRECT",
+		"ip6tables -t nat -I PREROUTING 1 -i eth0 -j RETURN", "ip6tables -t nat -I PREROUTING 1 -i eth1 -j RETURN",
+	}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Output mismatch.\nExpected: %#v\nActual: %#v", expected, actual)
+	}
+}
+
+func TestHandleInboundIpv6RulesWithIpNets(t *testing.T) {
+	cfg := constructConfig()
+	iptConfigurator := NewIptablesConfigurator(cfg)
+	iptConfigurator.cfg.EnableInboundIPv6s = net.IPv6loopback
+	_, ipnet, _ := net.ParseCIDR("10.0.0.0/8")
+	ipv6Range := NetworkRange{
+		IsWildcard: false,
+		IPNets:     []*net.IPNet{ipnet},
+	}
+	iptConfigurator.cfg.InboundPortsInclude = "4000,5000"
+	iptConfigurator.cfg.InboundPortsExclude = "4000,5000"
+	iptConfigurator.cfg.KubevirtInterfaces = "eth0,eth1"
+	iptConfigurator.handleInboundIpv6Rules(ipv6Range, ipv6Range)
+	actual := utils.FormatIptablesCommand(iptConfigurator.iptables.BuildV6())
+	expected := []string{
+		"ip6tables -t nat -N ISTIO_REDIRECT", "ip6tables -t nat -N ISTIO_IN_REDIRECT",
+		"ip6tables -t nat -N ISTIO_INBOUND", "ip6tables -t nat -N ISTIO_OUTPUT",
+		"ip6tables -t nat -A ISTIO_REDIRECT -p tcp -j REDIRECT --to-port 15001",
+		"ip6tables -t nat -A ISTIO_IN_REDIRECT -p tcp -j REDIRECT --to-port 15001",
+		"ip6tables -t nat -A PREROUTING -p tcp -j ISTIO_INBOUND",
+		"ip6tables -t nat -A ISTIO_INBOUND -p tcp --dport 4000 -j ISTIO_IN_REDIRECT",
+		"ip6tables -t nat -A ISTIO_INBOUND -p tcp --dport 5000 -j ISTIO_IN_REDIRECT",
+		"ip6tables -t nat -A OUTPUT -p tcp -j ISTIO_OUTPUT",
+		"ip6tables -t nat -A ISTIO_OUTPUT -o lo -s ::6/128 -j RETURN",
+		"ip6tables -t nat -A ISTIO_OUTPUT -o lo ! -d ::1/128 -j ISTIO_IN_REDIRECT",
+		"ip6tables -t nat -A ISTIO_OUTPUT -d ::1/128 -j RETURN",
+		"ip6tables -t nat -A ISTIO_OUTPUT -d 10.0.0.0/8 -j RETURN",
+		"ip6tables -t nat -I PREROUTING 1 -i eth0 -d 10.0.0.0/8 -j ISTIO_REDIRECT",
+		"ip6tables -t nat -I PREROUTING 1 -i eth1 -d 10.0.0.0/8 -j ISTIO_REDIRECT",
+		"ip6tables -t nat -A ISTIO_OUTPUT -d 10.0.0.0/8 -j ISTIO_REDIRECT",
+		"ip6tables -t nat -A ISTIO_OUTPUT -j RETURN"}
+	if !reflect.DeepEqual(actual, expected) {
+		t.Errorf("Output mismatch.\nExpected: %#v\nActual: %#v", expected, actual)
+	}
+}
+
+func TestHandleInboundIpv6RulesWithUidGid(t *testing.T) {
+	cfg := constructConfig()
+	iptConfigurator := NewIptablesConfigurator(cfg)
+	iptConfigurator.cfg.EnableInboundIPv6s = net.IPv6loopback
+	_, ipnet, _ := net.ParseCIDR("10.0.0.0/8")
+	ipv6Range := NetworkRange{
+		IsWildcard: false,
+		IPNets:     []*net.IPNet{ipnet},
+	}
+	iptConfigurator.cfg.InboundPortsInclude = "4000,5000"
+	iptConfigurator.cfg.InboundPortsExclude = "4000,5000"
+	iptConfigurator.cfg.KubevirtInterfaces = "eth0,eth1"
+	iptConfigurator.cfg.ProxyGID = "1,2"
+	iptConfigurator.cfg.ProxyUID = "1,2"
```

Comment:
Pick different IDs to make sure GID and UID are not mixed up:
```suggestion
	iptConfigurator.cfg.ProxyUID = "3,4"
```